### PR TITLE
NUTCH-2868 urlnormalizer-protocol fails with StringIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by sebastian-nagel**
... when reading invalid line in configuration file
- log invalid line and skip over it
- more verbose logging which configuration file is read

### CODE REVIEW

File: src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java [hunk 1]
```diff
@@ -82,15 +82,21 @@ private synchronized void readConfiguration(Reader configReader) throws IOExcept
...
+      lineNumber++;
       if (StringUtils.isNotBlank(line) && !line.startsWith("#")) {
         line = line.trim();
         delimiterIndex = line.indexOf(" ");
         // try tabulator
         if (delimiterIndex == -1) {
           delimiterIndex = line.indexOf("\t");
         }
+        if (delimiterIndex == -1) {
+          LOG.warn("Invalid line {} without delimiter: {}", lineNumber, line);
```
Comment (lewismc) [inline src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java:97]: How exactly do I trigger this warning? Can you provide an example or a trivial unit test?

**Review by lewismc — COMMENTED** [review]
LGTM however I am unsure how to reproduce the bug.

File: src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java [hunk 2]
```diff
@@ -177,6 +183,7 @@ public void setConf(Configuration conf) {
       if (reader == null) {
         Path path = new Path(file);
         FileSystem fs = path.getFileSystem(conf);
+        LOG.info("Reading {} rules file {} from {}", pluginName, file, fs);
```
[the remark this answers is not in the document]
Comment (sebastian-nagel) [reply src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java:186]: Yes, not really useful. Will change.

File: src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java (hunk 1 above)
Comment (sebastian-nagel) [reply src/plugin/urlnormalizer-protocol/src/java/org/apache/nutch/net/urlnormalizer/protocol/ProtocolURLNormalizer.java:97]: Seen with a line including no (or an empty) host name. But yes, I'll extend the unit tests.

**Comment by sebastian-nagel** [comment]
Thanks, @lewismc. All points adressed.

**Review by lewismc — APPROVED** [review]
+1

**Pull request merged**